torch/csrc/jit/symbolic_script.cpp

```diff
@@ -51,6 +354,24 @@ Argument originalReturnType(const TupleTypePtr& tup) {
   return Argument("", TupleType::create(std::move(types)));
 }
 
+// In torchscript AD formulas, we define {func_0, func_1, ...} as
+// overloaded functions of `func`.
+// Remove the suffix before adding the schema string to map
+// schema_to_graphs.
+std::string overloadedSchemaString(const FunctionSchema& schema) {
+  const auto& schema_name = schema.name();
+  auto pos = schema_name.find_last_of('_');
+  auto schema_name_suffix = schema_name.substr(pos + 1);
+  std::string schema_string = canonicalSchemaString(schema);
+  if (!schema_name_suffix.empty()
+      && schema_name_suffix.find_first_not_of("0123456789") == string::npos) {
```

Comment:
missing `std::` before `string::npos`, failing here on all my builds :/, weird that no CI catched this. @ailzhang

Reply:
@sinkingsugar  Would you mind using the collect_env.py (https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py) to let me know your environment? 
Basically this is indeed a typo on my side, but it got hidden by this header file included indirectly.  https://github.com/pytorch/pytorch/blob/master/c10/util/string_utils.h#L6 We likely want to fix this on our side.  But I'm curious why your build failed. Thanks!

Reply:
At a guess, probably some ifdefs toggled differently causing the header to not be included.

Reply:
@ezyang Yea agree, would love to know @sinkingsugar 's build environment, maybe worth adding to our CI?

Reply:
We can probably figure it out just by tracing the include path ourselves :)

Reply:
@ailzhang 
well, didn't have time to run the script but I will just give you the full Dockerfile 😄 
https://gist.github.com/sinkingsugar/81a42c6104bf01efd00e1d4ff9358c9b

Reply:
@sinkingsugar Interestingly the Dockerfile built successfully on my end :( 
![screen shot 2019-02-20 at 11 05 43 am](https://user-images.githubusercontent.com/5248122/53117374-9d303d80-34ff-11e9-9564-253bd1e14723.png)

Reply:
Hmm weird :) pretty sure I gave you the right one! but on master it's now fixed anyway so all good I suppose!